samza-azure/src/main/java/org/apache/samza/system/azureblob/avro/AzureBlobOutputStream.java

```diff
@@ -85,6 +85,7 @@
 
   private volatile boolean isClosed = false;
   private long totalUploadedBlockSize = 0;
+  private long totalNumberOfRecordsInBlob = 0;
```

Comment:
Was there any consideration given to using AtomicLong here instead? If threading is involved, it could more clearly express that any modifications will be atomic without the need to worry about method synchronization.

Reply:
AtomicLong is a nice thing to do. However, the methods of this class are all synchronized as there are other aspects beyond this variable that require single threaded access.

Reply:
Cool. That's fine then.